src/constraint/spec.ts

```diff
@@ -101,6 +101,49 @@ export const SPEC_CONSTRAINTS: SpecConstraintModel[] = [
     }
   },
   {
+    name: 'alwaysIncludeZeroInScaleWithBarMark',
+    description: 'Do not reccommend bar mark if scale does not start at zero',
+    properties: [Property.MARK, Property.SCALE, Property.SCALE_ZERO, Property.CHANNEL, Property.TYPE],
+    requireAllPropertiesSpecific: true,
+    strict: true,
+    satisfy: (specM: SpecQueryModel, schema: Schema, opt: QueryConfig) => {
+      const mark = specM.getMark();
+      const encodings = specM.getEncodings();
+
+      if (mark === Mark.BAR) {
+        for (let encQ of encodings) {
+          if ((encQ.channel === Channel.X || encQ.channel === Channel.Y) && encQ.scale && encQ.type === Type.QUANTITATIVE) {
```

Comment:
This `if` is hard to read.  You might want to use contain and put predicates on 3-4 lines.